### PR TITLE
Login Page UI changes

### PR DESCRIPTION
**Pull request by rupalg**
This is the new UI for the log in page for the Flutter app.
Screenshot 2023-07-06 at 8 06 27 PM

I need feedback on the following:
1) I used an external package for the social icons called "FlutterSocialButton": is that recommended? If not, I can use some external image icons instead. 
2) How do you localize text fields within the form, for instance: "Email or phone number" or "password"? The standard l10n.field_name is giving an error, namely "201:52: Error: Not a constant expression."

### CODE REVIEW

File: lib/l10n/app_en.arb [hunk 1]
```diff
@@ -39,5 +39,14 @@
     "reminderBannerProfileUpdateTitle": "Update your profile",
     "reminderBannerProfileUpdateSubtitle": "Make sure your profile is updated so you get the best matches",
     "reminderBannerProfileUpdateCta": "Update profile",
-    "recommendedMentorCardScrollable": "Recommended"
+    "recommendedMentorCardScrollable": "Recommended",
+    "welcome": "Welcome!",
+    "signInMessage": "Please sign in your account with",
```
Comment (shadowbrush) [inline lib/l10n/app_en.arb:44]: Please sign **into** your account with

Comment (rupalg) [reply lib/l10n/app_en.arb:44]: Great catch, resolved.

**Comment by rupalg** [comment]
> Would it be possible to left align the social sign-in items (Google, Facebook, ...)?

Yes, here's what it looks like. What do you think?

Screenshot 2023-07-06 at 8 51 01 PM

**Comment by andersan** [comment]
The wireframe had these buttons center-aligned, so that feedback likely needs to be directed to @katymitchell-mm https://trello.com/c/WhNROPMC/53-p0-login-page

**Comment by shadowbrush** [comment]
@katymitchell-mm If you agree with the change from "Welcome!" to "Welcome Back!", please indicate it here. Thanks!

**Comment by katymitchell-mm** [comment]
This looks great!

I'm good with "Welcome back!" 

Is it possible to keep the left alignment of the social icons but center the whole block of them on the page?

**Comment by shadowbrush** [comment]
login-screen

**Comment by katymitchell-mm** [comment]
perfect!

**Comment by rupalg** [comment]
Made all changes, here's the final UI. Requesting review and approval.
Screenshot 2023-07-11 at 6 58 04 PM

**Comment by ashlilili** [comment]
Overall looks great! Thanks Rupal for working on this!
I don't have preference between the flutter icon package you are using and actual icons - but is there a way to make them smaller to match the size of the wireframe design?
Another nit is can we make the Welcome bigger and the "please sign in to your account with" smaller?

**Comment by rupalg** [comment]
Hi Ashley! Thanks for the feedback. 

I have removed the flutter icon package and used images instead because I realized that the flutter icons size are not adjustable. 

I have increased the size of the "Welcome" size and the rest of the profile elements. 

Here's what it looks like now:

Screenshot 2023-07-11 at 11 53 57 PM

File: lib/widgets/screens/sign_in/sign_in_screen.dart [hunk 2]
```diff
@@ -64,161 +65,342 @@ class _SignInScreenState extends State<SignInScreen> {
...
-                                  'Sign In',
-                                  style: GoogleFonts.poppins(
-                                    fontSize: 14,
-                                    fontWeight: FontWeight.bold,
+                                },
+                                child: Padding(
+                                  padding: const EdgeInsets.symmetric(
+                                    vertical: Insets.widgetSmallInset,
+                                    horizontal: Insets.widgetLargeInset * 2.5,
+                                  ),
```
Comment (m-rosario) [inline lib/widgets/screens/sign_in/sign_in_screen.dart:213]: We should avoid performing math operations on constants that will result in another constant. This might make sense, for example, when one widget is sized relative to another, but in this case 2.5xlargeInset seems like arbitrary value.

File: lib/widgets/screens/sign_in/sign_in_screen.dart [hunk 3]
```diff
@@ -64,161 +65,342 @@ class _SignInScreenState extends State<SignInScreen> {
...
+                            color: Colors.black,
+                          ),
+                          Text(
+                            l10n.or,
+                            style: TextStyles.sectionSubNote(context),
+                          ),
+                          Container(
+                            width: Dimensions.loginBoxWidth / 2,
+                            height: Dimensions.lineHeight,
+                            color: Colors.black,
```
Comment (m-rosario) [inline lib/widgets/screens/sign_in/sign_in_screen.dart:255]: Use the colorScheme colors instead of hardcoding them.

File: lib/widgets/screens/sign_in/sign_in_screen.dart [hunk 4]
```diff
@@ -64,161 +65,342 @@ class _SignInScreenState extends State<SignInScreen> {
...
+                    Padding(
+                      padding: const EdgeInsets.all(Insets.widgetSmallInset),
+                      child: Row(
+                        mainAxisAlignment: MainAxisAlignment.center,
+                        children: [
+                          Container(
+                            width: Dimensions.loginBoxWidth / 2,
+                            height: Dimensions.lineHeight,
+                            color: Colors.black,
+                          ),
```
Comment (m-rosario) [inline lib/widgets/screens/sign_in/sign_in_screen.dart:247]: It might be easier to use a Divider widget. Also you can play with flex values to make the widgets occupy all of the available horizontal space, instead of using a constant value.

File: lib/widgets/screens/sign_in/sign_in_screen.dart [hunk 5]
```diff
@@ -64,161 +65,342 @@ class _SignInScreenState extends State<SignInScreen> {
...
+                              _openSignUpScreen(context);
+                            },
+                            child: Row(
+                              mainAxisSize: MainAxisSize.min,
+                              children: [
+                                const SizedBox(
+                                  width: Dimensions.iconWidth,
+                                  child: Image(
+                                    image:
+                                        AssetImage('assets/images/google.png'),
```
Comment (m-rosario) [inline lib/widgets/screens/sign_in/sign_in_screen.dart:280]: Define the asset in the constants library: paths.dart / Assets

File: lib/widgets/atoms/user_search_bar.dart [hunk 6]
```diff
@@ -11,6 +11,17 @@ class UserSearchBar extends StatelessWidget {
...
+      backgroundColor: MaterialStateProperty.all(
+          Theme.of(context).colorScheme.primaryContainer),
+      padding: MaterialStateProperty.all(
+        const EdgeInsets.fromLTRB(
+            Insets.widgetLargeInset,
+            Insets.widgetSmallInset,
+            Insets.widgetSmallInset,
+            Insets.widgetSmallInset),
+      ),
+      // shape: MaterialStateProperty.all({BorderSide side: BorderSide.none,}),
```
Comment (m-rosario) [inline lib/widgets/atoms/user_search_bar.dart:24]: I don't see this Widget being used anywhere else in the PR. Why is this change included?

File: lib/constants/dimensions.dart [hunk 7]
```diff
@@ -43,6 +43,12 @@ class Dimensions {
   static const double mentorCardWidth = 260;
   static const double chipVisualDensity = -4;
   static const double chipMaxWidth = 100;
+  static const double loginBoxWidth = 360;
+  static const double lineHeight = 1;
+  static const double textBoxWidth = 200;
+  static const double sizedBoxWidth = 360;
+  static const double iconWidth = 20;
+  static const double iconSpaceWidth = 10;
```
Comment (m-rosario) [inline lib/constants/dimensions.dart:51]: Whenever possible, widths and heights should be multiples of 8 (preferred), or multiple of 4 (for smaller components). This so that elements align nicely in an imaginary grid. Check out this for more details: https://m2.material.io/design/layout/spacing-methods.html

**Review by m-rosario — CHANGES_REQUESTED** [review]
Looking good! Left some comments in the code that I think should be addressed before approving.

Something else: I think it would also look better if the vertical spacing between components is more consistent. It feels like some components have too much spacing and others too little, but this is just a nit and we can always iterate later.

File: lib/widgets/screens/sign_in/sign_in_screen.dart [hunk 8]
```diff
@@ -64,161 +65,342 @@ class _SignInScreenState extends State<SignInScreen> {
...
+                          crossAxisAlignment: CrossAxisAlignment.center,
+                          children: [
+                            SizedBox(
+                              height: mediaQuery.height * 0.03,
+                            ),
+                            TextFormField(
+                                key: const Key('emailTextField'),
+                                controller: emailController,
+                                decoration: const InputDecoration(
+                                  labelText: "Email or phone number",
```
[the remark this answers is not in the document]
Comment (rupalg) [reply lib/widgets/screens/sign_in/sign_in_screen.dart:130]: Hi Miguel. Great catch. I have mentioned this issue in the PR as well noted in this question: How do you localize text fields within the form, for instance: "Email or phone number" or "password"? The standard l10n.field_name is giving an error, namely "201:52: Error: Not a constant expression." Would love to hear if you have any thoughts. :)

File: lib/widgets/screens/sign_in/sign_in_screen.dart [hunk 9]
```diff
@@ -64,161 +65,342 @@ class _SignInScreenState extends State<SignInScreen> {
...
+                                  border: OutlineInputBorder(),
+                                ),
+                                validator: (value) {
+                                  bool validEmail = _validateEmail(value!);
+                                  if (validEmail != true) {
+                                    return 'Please enter a valid email or phone number';
+                                  }
```
[the remark this answers is not in the document]
Comment (rupalg) [reply lib/widgets/screens/sign_in/sign_in_screen.dart:137]: Note the comment above^

File: lib/widgets/screens/sign_in/sign_in_screen.dart (hunk 2 above)
Comment (rupalg) [reply lib/widgets/screens/sign_in/sign_in_screen.dart:213]: Thanks for that comment. I'll create a new constant.

File: lib/widgets/screens/sign_in/sign_in_screen.dart (hunk 3 above)
Comment (rupalg) [reply lib/widgets/screens/sign_in/sign_in_screen.dart:255]: Thanks for the comment. I'll make that change.

File: lib/widgets/screens/sign_in/sign_in_screen.dart (hunk 5 above)
Comment (rupalg) [reply lib/widgets/screens/sign_in/sign_in_screen.dart:280]: Awesome, will do!

File: assets/images/facebook.png
[the remark this answers is not in the document]
Comment (rupalg) [reply assets/images/facebook.png:1]: Ah, great point. Thanks Miguel. Do you have guidance on what size the images should be (max sizes?). I can resize this asset according to those guidelines.

File: lib/widgets/screens/sign_in/sign_in_screen.dart (hunk 4 above)
Comment (rupalg) [reply lib/widgets/screens/sign_in/sign_in_screen.dart:247]: Hey Miguel, 
The reason I didn't use the divider widget is because the divider widget takes the whole row. In this scenario, there are two horizontal lines separated with the word "or". I will try to play around with the divider to see it the rendering makes sense, and comment here.

File: lib/widgets/atoms/user_search_bar.dart (hunk 6 above)
Comment (rupalg) [reply lib/widgets/atoms/user_search_bar.dart:24]: Ah, great catch! This looks like I may have merged the widget from the other branch I'm working on for the search bar on the explore page. I will remove this widget from this PR. Thanks!

File: lib/constants/dimensions.dart (hunk 7 above)
Comment (rupalg) [reply lib/constants/dimensions.dart:51]: Ah, fantastic. This is a great read. I'll read up more on material design, and make these changes.

File: assets/images/facebook.png
[the remark this answers is not in the document]
Comment (m-rosario) [reply assets/images/facebook.png:1]: The best way to do this is to create different sizes for different screen densities. That is well outside the scope of this PR so we can just leave this as is for now and I can take care of creating the variable assets later :)

File: lib/widgets/screens/sign_in/sign_in_screen.dart (hunk 8 above)
[the remark this answers is not in the document]
Comment (m-rosario) [reply lib/widgets/screens/sign_in/sign_in_screen.dart:130]: Oh I missed that! That happens because you have 'const' right before InputDecoration, and l10n.field_name is not a constant. When you declare an object as constant, you are basically saying that every field under it is also constant. The solution here is to remove 'const' from InputDecoration and apply it only to the subfields that are constant, which won't include labelText. Example:

```
...
decoration: InputDecoration(
        labelText: l10n.field_name,
        border: const OutlineInputBorder(),
),
...
```

File: assets/images/facebook.png
[the remark this answers is not in the document]
Comment (rupalg) [reply assets/images/facebook.png:1]: Awesome, thanks so much. Leaving this for now.

File: lib/widgets/screens/sign_in/sign_in_screen.dart (hunk 8 above)
[the remark this answers is not in the document]
Comment (rupalg) [reply lib/widgets/screens/sign_in/sign_in_screen.dart:130]: Ah, that makes sense. Thanks, Miguel. Will make changes accordingly.

File: lib/widgets/screens/sign_in/sign_in_screen.dart (hunk 2 above)
Comment (rupalg) [reply lib/widgets/screens/sign_in/sign_in_screen.dart:213]: Made changes by creating new constants.

File: lib/widgets/atoms/user_search_bar.dart (hunk 6 above)
Comment (rupalg) [reply lib/widgets/atoms/user_search_bar.dart]: I have removed the file from the PR.

**Comment by rupalg** [comment]
Hello Miguel, 
I have made the requested changes. You'll see multiple commits above as I ran into multiple merge conflicts post the GraphQL codegen migration. But, hopefully, things work now. Here's a screenshot of the login page. 

Screenshot 2023-07-13 at 7 41 08 PM

**Comment by andersan** [comment]
Congrats on the merged PR, Rupal! I haven't been following along that close but it looks like this is one of your first.
